scheduler/core/programprovider/gpp/gppprogramprovider.py

```diff
@@ -0,0 +1,1333 @@
+        name, magnitudes, target_type = self._parse_target_header(data)
+        ra_hhmmss = data[GppProgramProvider._TargetKeys.RA]
+        dec_ddmmss = data[GppProgramProvider._TargetKeys.DEC]
+
+        # TODO: Is this the proper way to handle conversions from hms and dms?
+        ra = sex2dec(ra_hhmmss, to_degree=True)
+        dec = dmsstr2deg(dec_ddmmss)
+
+        pm_ra = data.setdefault(GppProgramProvider._TargetKeys.DELTA_RA, 0.0)
+        pm_dec = data.setdefault(GppProgramProvider._TargetKeys.DELTA_DEC, 0.0)
+        epoch = data.setdefault(GppProgramProvider._TargetKeys.EPOCH, 2000)
+
+        return SiderealTarget(
+            name=name,
+            magnitudes=frozenset(magnitudes),
+            type=target_type,
+            ra=ra,
+            dec=dec,
+            pm_ra=pm_ra,
+            pm_dec=pm_dec,
+            epoch=epoch)
+
+    def parse_nonsidereal_target(self, data: dict) -> NonsiderealTarget:
+        """
+        TODO: Retrieve the Ephemeris data.
+        TODO: Should we be doing this here, or in the Collector?
+        """
+        name, magnitudes, target_type = self._parse_target_header(data)
+        des = data.get(GppProgramProvider._TargetKeys.DES, name)
+
+        # This is redundant: it is always 'nonsidereal'
+        # tag = data[GppProgramProvider._TargetKeys.TAG]
+
+        # This is the tag information that we want: either MAJORBODY, COMET, or ASTEROID
+        tag_str = data[GppProgramProvider._TargetKeys.NONSIDEREAL_OBJECT_TYPE]
+        tag = TargetTag[tag_str]
+
+        # RA and dec will be looked up when determining target info in Collector.
+        return NonsiderealTarget(
+            name=name,
+            magnitudes=frozenset(magnitudes),
+            type=target_type,
+            des=des,
+            tag=tag)
+
+    @staticmethod
+    def _parse_instrument(data: dict) -> str:
+        """Get the instrument name"""
+        fpu = None
+        instrument = None
+
+        for step in data:
+            # Ignore acquisitions
+            if step[GppProgramProvider._AtomKeys.OBS_CLASS].upper() not in [ObservationClass.ACQ.name,
+                                                                            ObservationClass.ACQCAL.name]:
+                inst_key = GppProgramProvider._AtomKeys.INSTRUMENT
+                # Visitor instrument names are in GppProgramProvider._AtomKeys.INST_NAME
+                if GppProgramProvider._AtomKeys.INST_NAME in step:
+                    inst_key = GppProgramProvider._AtomKeys.INST_NAME
+                instrument = step[inst_key].split(' ')[0]
+
+                if instrument in GppProgramProvider.FPU_FOR_INSTRUMENT:
+                    if GppProgramProvider.FPU_FOR_INSTRUMENT[instrument] in step:
+                        fpu = step[GppProgramProvider.FPU_FOR_INSTRUMENT[instrument]]
+
+                if instrument == 'GMOS-N' and fpu == 'IFU Left Slit (blue)':
+                    instrument = 'GRACES'
```

Comment:
This seems a bit odd, how does it become `GRACES`?

Reply:
This is left over from the OCS. We won't need to support this in GPP.